moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -108,6 +110,59 @@ void MotionPlanningFrame::onClearOctomapClicked()
   clear_octomap_service_client_.call(srv);
 }
 
+bool MotionPlanningFrame::computeCartesianPlan()
+{
+  ros::WallTime start = ros::WallTime::now();
+  // get goal pose
+  robot_state::RobotState goal = *planning_display_->getQueryGoalState();
+  std::vector<geometry_msgs::Pose> waypoints;
+  const std::string& link_name = move_group_->getEndEffectorLink();
+  const robot_model::LinkModel* link = move_group_->getRobotModel()->getLinkModel(link_name);
+  if (!link)
+  {
+    ROS_ERROR_STREAM("Failed to determine unique end-effector link: " << link_name);
+    return false;
+  }
+  waypoints.push_back(tf2::toMsg(goal.getGlobalLinkTransform(link)));
+
+  // setup default params
+  double cart_step_size = 0.01;
+  double cart_jump_thresh = 0.0;
+  bool avoid_collisions = true;
+
+  // compute trajectory
+  moveit_msgs::RobotTrajectory trajectory;
+  double fraction =
+      move_group_->computeCartesianPath(waypoints, cart_step_size, cart_jump_thresh, trajectory, avoid_collisions);
+
+  if (fraction >= 1.0)
```

Comment:
This used to be `> 0`. With this version, the button fails silently. Is that on purpose?

Reply:
Yes, this was on purpose: If you cannot reach your target, the planning request should fail.
What do you mean by "fails silently"? It should be written "FAILED" and the partial trajectory should be shown.

Reply:
My bad, it will display "Failed" in the UI. I don't see how the partial trajectory would be displayed, but I'll go test first.